### PR TITLE
Feat/add abort action to chat assistant

### PR DESCRIPTION
**Pull request by cephaschapa**
Changes
- Adds abort controller to manage text generation
- Adds stop button with stop generation handler
- Adjusts chat assistant proportions
- Adds state to manage buttons

### CODE REVIEW

File: app/src/app/api/v0/assistants/threads/messages/route.ts [hunk 1]
```diff
@@ -22,9 +26,28 @@ export const POST = apiHandler(async (req) => {
...
+      threadId,
+      {
+        assistant_id: assistantId,
+      },
+      {
+        signal,
+      },
+    );
+
+    // TODO: prevent a new thread from being added to current run when active
```
Comment (lemilonkh) [inline app/src/app/api/v0/assistants/threads/messages/route.ts:41]: Do we have to fix this before merging?

File: app/src/app/api/v0/assistants/threads/messages/route.ts [hunk 2]
```diff
@@ -22,9 +26,28 @@ export const POST = apiHandler(async (req) => {
...
+
+    return new NextResponse(stream.toReadableStream());
+  } catch (error: any) {
+    if (error.name === "AbortError") {
+      console.log("OpenAI API request was aborted");
```
Comment (lemilonkh) [inline app/src/app/api/v0/assistants/threads/messages/route.ts:46]: Should we remove this? Seems like a common operation.

File: app/src/components/ChatBot/chat-bot.tsx [hunk 3]
```diff
@@ -591,14 +637,23 @@ export default function ChatBot({
...
-            color="content.tertiary"
-            aria-label="Send message"
-            isDisabled={inputDisabled}
-          />
+          {inputDisabled ? (
+            <IconButton
+              onClick={stopGeneration}
+              icon={<MdStop />}
+              colorScheme="red"
+              aria-label="Stop generation"
```
Comment (lemilonkh) [inline app/src/components/ChatBot/chat-bot.tsx:645]: I18n

File: app/src/components/ChatBot/chat-bot.tsx [hunk 4]
```diff
@@ -591,14 +637,23 @@ export default function ChatBot({
...
+            />
+          ) : (
+            <IconButton
+              type="submit"
+              variant="ghost"
+              icon={<MdOutlineSend size={24} />}
+              color="content.tertiary"
+              aria-label="Send message"
```
Comment (lemilonkh) [inline app/src/components/ChatBot/chat-bot.tsx:653]: I18n

**Review by lemilonkh — APPROVED** [review]
Thanks ✨

File: app/src/app/api/v0/assistants/threads/messages/route.ts (hunk 1 above)
Comment (cephaschapa) [reply app/src/app/api/v0/assistants/threads/messages/route.ts:41]: I need to do a spike on this, we are currently using a version of the sdk that has some of its methods that are depricated. I create a ticket to address that.

**Pull request merged**